### PR TITLE
Add Vue.js as the new JavaScript framework for the UI and remove Mithril.js

### PR DESCRIPTION
**Pull request by dustinryerson**
This is a major rewrite of the UI, replacing Mithril.js with Vue.js as the frontend framework. The rewrite is almost finished - I'm going through final rounds of testing and hunting down the last few bugs that I can find.

I have a running task list of work that I'd like to do, which includes:
1. Full test coverage (or as much as possible)
2. Add detailed code comments and docs
3. Establish linting guidelines
4. Remove the dependency on the Bulma Swatch theme (but add all the necessary styling from https://github.com/jenil/bulmaswatch/tree/gh-pages/superhero)
5. Integrate some helpful apps:
  a. Automated Dependency Management: [Greenkeeper](https://greenkeeper.io/)
  b. Test Coverage Reporting: [Codecov](https://codecov.io/) or [Coveralls](https://coveralls.io/)
  c. Error Tracking/Crash Reporting & Monitoring: [Rollbar](https://rollbar.com/) or [Bugsnag](https://www.bugsnag.com/)

### CODE REVIEW

**Comment by perigrin** [comment]
We use Rollbar for error reporting on the backend so probably be a good choice there.

File: src/App.vue [hunk 1]
```diff
@@ -0,0 +1,158 @@
+<template>
+    <div id="app">
+        <SignIn v-if="this.$route.path === '/'" />
+        <div v-else-if="this.$route.params.currentWorkspace || this.$route.path === '/user'">
```
Comment (dustinryerson) [inline src/App.vue]: This condition doesn't seem sustainable as pages that don't require the `currentWorkspace` parameter are added. I'll push up a commit better solution.

Comment (perigrin) [reply src/App.vue]: Yeah long term I think you're right but for now I wouldn't let this stop me from merging this.

File: src/views/SignIn/__tests__/SignIn.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        // how to handle getters that take arguments. It seems to identify such
+        // calls as method calls and looks for a corresponding item in the
+        // 'methods' object of the Vue component.
+        Object.defineProperty(wrapper.vm, 'loadCurrentWorkspace', { value: jest.fn() });
+
+        wrapper.setData({ emailAddress: 'user@joyent.com', password: 'abcdefg' });
+        wrapper.find('button').trigger('click');
+
+        expect(spy).toHaveBeenCalledWith({ user: 'user@joyent.com', password: 'abcdefg' });
+    });
```
Comment (perigrin) [inline src/views/SignIn/__tests__/SignIn.test.js]: We should probably have a test for `isForcePasswordChange()` being true in here too.

**Pull request merged**